### ISSUE TITLE
DetailsList - not able to reorder columns via keyboard

### ISSUE THREAD

**Issue by haixiehx** [microsoft/fluentui#19588]
### Environment Information

- **Package version(s)**: (fill this out)
- **Browser and OS versions**: (fill this out)

### Describe the issue:

I am able to reorder columns use mouse but not able to reorder via keyboard. If I was a keyboard user, I am not able to reorder columns on DetailsList.

### Please provide a reproduction of the issue in a codepen:
https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/draganddrop

#### Actual behavior:

Keyboard users are not able to reorder columns on DeatilsList.

#### Expected behavior:

Keyboard users should be able to reorder columns and screen readers should announce drag and drop actions properly.
There should be some usage instructions and provide the abilities to custom messages:
- "Press <someKey> to start a drag. When dragging you can use the arrow keys to move the item around and escape to cancel...",
- "You have lifted an item ...",
- "You have moved the item <name> from position ... to ...",
- etc.

**Comment by gouttierre** [comment]
@haixiehx - Thanks for filing this issue with us. To set expectation our team will have a look at the issue but only once capacity allows.

@bsunderhus -  Would you be able to confirm if this is a regression, or if this behavior is an issue?
cc @smhigley

**Comment by bsunderhus** [comment]
This is not a regression neither an issue, it's not currently supported on `DetailsList`, there's no mentioning on drag&drop a11y support on the component itself or in the `DragDropHelper` which is used internally, if I'm not mistaken (@smhigley, can you confirm?).

This is a definitely a nice to have feature though! I'll be adding this to backlog to look for it in the future.

**Comment by ecraig12345** [comment]
This is a duplicate of #18788.
